include/seastar/core/metrics_api.hh

```diff
@@ -240,8 +239,9 @@ using metrics_registration = std::vector<register_ref>;
 
 class metric_groups_impl : public metric_groups_def {
     metrics_registration _registration;
+    shared_ptr<impl> _impl; // keep impl alive while metrics are registered
 public:
-    metric_groups_impl() = default;
+    metric_groups_impl();
```

Comment:
This looks like a leftover from some mid-code changes; the contractor still uses the default implementation.

Reply:
Yes you are right. I had this at first when I tried initing the impl SP in the constructor directly but then had to move it because of the bug mentioned in the comment.